src/tip-opengov.ts

```diff
@@ -127,12 +138,13 @@ async function signAndSendCallback(
 }
 
 const tryGetReferendumId = async (
-  api: ApiPromise,
+  api: API,
   blockHash: string,
   encodedProposal: string,
   log: Probot["log"],
 ): Promise<null | number> => {
   try {
+    // TODO: Convert
     const referendumId = await getReferendumId(await api.at(blockHash), encodedProposal);
```

Comment:
I see you found a different way, right?

Reply:
I found that I obtain all the events (check out my changed in the `getReferendumId`) but I can't compare the event to the encoded proposal, so I don't know if it's fine.

Reply:
[This is what I'm doing](https://github.com/paritytech/substrate-tip-bot/pull/158/files#diff-3294a832ea2276e554177e0b3007cc2d401c082912c7fbde49fa09141bf1aed1R189)

But the object only contains the following:
```typescript
{
    meta: {
        block: BlockInfo;
        phase: EventPhase;
    };
    payload: {
        track: number;
        index: number;
        proposal: PreimagesBounded;
    };
}
```

I have trouble comparing the data to the encoded proposal. Can I compare it to the index or some other information instead?

Reply:
Hmm, `index` is the information you want to find out in this function.

Worst case, we can accept the most recent referendum on our track. Or read a latest referendum number before creating ours, and take `x+1` as our index - in fact this was the first version here.

There is a risk that someone else creates a referendum and we mess up our count - that's why we compared the proposal to make sure.

Reply:
I'm getting the referendum in a specific block. It could happen that two referendums are submitted at the same time, which would be troublesome.

but I don't know how to compare the data which would be the safest choice 🤔

Reply:
Doesn't this `PreimagesBounded` contain what we need?